apps/shell/views/index.hbs

```diff
@@ -35,6 +35,13 @@
   window.onload = function() {
     lib.init(setupHterm);
   };
+
+  var urlWithToken = function(){
+    url = new URL(document.URL.replace(/^http/, 'ws').replace(/#.*$/, ''));
+    url.searchParams.append('csrf', '{{csrfToken}}');
+    return url;
+  }
```

Comment:
We added a polyfill for URLSearchParams to NoVNC. We should add that same polyfill for client side in this app.

Reply:
Remember to specify var or let here for the `url`. Also note that document.URL is a string whereas the urlWithToken is returning a URL object, but I guess that doesn't matter if it still connects successfully. Finally, you can use the same URL interface to set the protocol, instead of using replace.

```js
var urlWithToken = function(){
  let url = URL(document.URL.replace(/#.*$/, ''));
  url.protocol = 'ws:';
  url.searchParams.append('csrf', '{{csrfToken}}');
  return url;
}
```

Reply:
The replacement accounts for http(s) -> ws(s) though, so there'd have to be extra logic to figure out if we started out secure or not.

Reply:
To the polyfil question, we only use that so URLs like `/pun/sys/shell/ssh?foo=bar` work, where any query parameter other than csrf are not looked at.

Instead of a polyfil how about something like this?
```javascript
  var urlWithToken = function(){
    var base = document.URL.replace(/^http/, 'ws').replace(/#.*$/, '');
    var separator = '?'
    if(base.includes('?')){
      separator = '&'
    }
    return base + separator + "csrf={{csrfToken}}";
  }
```